### PR TITLE
Cli legacy alert

### PR DESCRIPTION
**Pull request by timngyn**
#### Description of changes:
- Updated the cli migration alert to use a warning callout
- Added a custom hook to get the current platform

Staging site: https://cli-legacy-alert.d1ywzrxfkb9wgg.amplifyapp.com/

Example links:
- https://cli-legacy-alert.d1ywzrxfkb9wgg.amplifyapp.com/javascript/tools/cli-legacy/graphql-transformer/client-codegen/
- https://cli-legacy-alert.d1ywzrxfkb9wgg.amplifyapp.com/javascript/tools/cli-legacy/graphql-transformer/function-directive/
- https://cli-legacy-alert.d1ywzrxfkb9wgg.amplifyapp.com/javascript/tools/cli/graphqlapi/client-code-generation/

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/utils/useCurrentPlatform.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Platform } from '@/data/platforms';
+import { useRouter } from 'next/router';
+
+// Custom hook to return the current platform
+export function useCurrentPlatform() {
+  const router = useRouter();
+
+  return router.query.platform as Platform;
```
Comment (reesscot) [inline src/utils/useCurrentPlatform.ts:8]: Jacob confirmed this should work in Next.js's SSG mode

**Review by hbuchel — APPROVED** [review]
Component LGTM. Thanks for the cleanup!

**Review by reesscot — APPROVED** [review]
Approved pending double-checking this doesn't cause hydration error warnings

File: src/utils/useCurrentPlatform.ts (hunk 1 above)
Comment (jacoblogan) [reply src/utils/useCurrentPlatform.ts:8]: yeah this is working with ssg, I'm seeing the link being generated correctly

**Pull request merged**